assets/questions/bstClosestValue.js

```diff
@@ -0,0 +1,216 @@
+      {
+        content:
+        [
+          'The tree is represented as a preorder traversal sequence of node' +
+          ' values in which \'#\'',
+          'represents an empty node. For example: the input \'2 1 # # #\'' +
+          ' represents the serialized',
+          'format of a binary tree using preorder order traversal, where' +
+          ' root value is 2 and only has a left child with value 1.'
+        ].join('\n'),
+        type: 'text'
+      },
+      {
+        content:
+        [
+          '#Definition for a binary tree node.',
+          'class TreeNode(object):',
+          '  def __init__(self, x):',
+          '    self.val = x',
+          '    self.left = None',
+          '    self.right = None'
+        ].join('\n'),
+        type: 'code'
+      }
+    ],
+    prerequisiteSkills: ['Binary Search Tree'],
+    acquiredSkills: ['Binary Search Tree', 'Recurssion', 'Iteration'],
+    inputFunctionName: 'AuxiliaryCode.deserialize',
+    outputFunctionName: null,
+    mainFunctionName: 'findClosestValue',
+    correctnessTests: [{
+      input: '2 1 # # 3 # #|1.1',
+      allowedOutputs: [1],
+      tag: 'simple BST'
+    },
+    {
+      input: '10 3 # 6 4 # # 7 # # #|4.1',
+      allowedOutputs: [4],
+      tag: 'result smaller than target'
+    },
+    {
+      input: '10 3 # # 14 # 15 14.7 # # 100 # #|14.6',
+      allowedOutputs: [14.7],
+      tag: 'result larger than target'
+    },
+    {
+      input: '2.5 # #|3.2',
+      allowedOutputs: [2.5],
+      tag: 'BST with one node'
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.failToUseAbsoluteDiff',
+      messages: [
+        "Run your code on '2 1 # # 3 # #|2.9' in your head, what is the" +
+        "expected result?",
+        "It looks like the issue is how you represent the difference" +
+        "between values",
+        [
+          "Do you use the absolute difference to measure closeness?"
+        ].join('')
+      ]
+    }],
+    performanceTests: []
+  }, {
+    instructions: [
+      {
+        content:
+        [
+          'Next, make sure you compare all possible candidates'
+        ].join(''),
+        type: 'text'
+      }
+    ],
+    prerequisiteSkills: ['Binary Search Tree'],
+    acquiredSkills: ['Binary Search Tree', 'Recurssion', 'Iteration'],
+    inputFunctionName: 'AuxiliaryCode.deserialize',
+    outputFunctionName: null,
+    mainFunctionName: 'findClosestValue',
+    correctnessTests: [{
+      input: '24 22 # # 25 # #|23.8',
+      allowedOutputs: [24],
+      tag: 'should return root'
+    },
+    {
+      input: '24 22 # # 25 # #|24.1',
+      allowedOutputs: [24],
+      tag: 'should return root'
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.failToCompareAll',
+      messages: [
+        "Run your code on '2 1 # # 3 # #|2.1' in your head, what is the" +
+        "expected result?",
+        "It looks like the issue is whether or not you consider all possible" +
```

Comment:
"It looks like your code doesn't correctly consider all possible candidate nodes."